OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs

```diff
@@ -101,8 +104,8 @@ public override void DoImpact(Target target, Actor firedBy, IEnumerable<int> dam
 				palette += firedBy.Owner.InternalName;
 
 			var explosion = Explosions.RandomOrDefault(Game.CosmeticRandom);
-			if (explosion != null)
-				world.AddFrameEndTask(w => w.Add(new Explosion(w, pos, explosion, palette)));
+			if (Image != null && explosion != null)
```

Comment:
I can't point to the line in `FieldLoader` where it is decided, but my experience is that it returns `null` instead of empty strings in all relevant cases.